dbt/include/cratedb/macros/adapters.sql

```diff
@@ -0,0 +1,112 @@
+  {% if contract_config.enforced and (not temporary) -%}
+      {{ get_table_columns_and_constraints() }} ;
+    insert into {{ relation }} (
+      {{ adapter.dispatch('get_column_names', 'dbt')() }}
+    )
+    {%- set sql = get_select_subquery(sql) %}
+  {% else %}
+    as
+  {% endif %}
+  (
+    {{ sql }}
+  );
+{%- endmacro %}
+
+{% macro cratedb__drop_relation(relation) -%}
+  {% call statement('drop_relation', auto_begin=False) -%}
+    drop {{ relation.type }} if exists {{ relation.render() }}
+  {%- endcall %}
+{% endmacro %}
+
+{% macro cratedb__truncate_relation(relation) -%}
+  {% call statement('truncate_relation') -%}
+    delete from {{ relation }}
+  {%- endcall %}
+{% endmacro %}
+
+{% macro cratedb__create_schema(relation) -%}
+  {% if relation.database -%}
+    {{ adapter.verify_database(relation.database) }}
+  {%- endif -%}
+  {%- call statement('create_schema') -%}
+  {# create schema if not exists {{ relation.without_identifier().include(database=False) }} #}
+    SELECT 1
+  {%- endcall -%}
+{% endmacro %}
+
+{% macro cratedb__drop_schema(relation) -%}
+  {% if relation.database -%}
+    {{ adapter.verify_database(relation.database) }}
+  {%- endif -%}
+  {%- call statement('drop_schema') -%}
+  {# drop schema if exists {{ relation.without_identifier().include(database=False) }} cascade #}
+    SELECT 1
+  {%- endcall -%}
+{% endmacro %}
+
+
+{# CrateDB: `COMMENT ON` not supported. #}
+{% macro cratedb__alter_relation_comment(relation, comment) %}
+  {% set escaped_comment = postgres_escape_comment(comment) %}
+  {% if relation.type == 'materialized_view' -%}
+    {% set relation_type = "materialized view" %}
+  {%- else -%}
+    {%- set relation_type = relation.type -%}
+  {%- endif -%}
+  {# comment on {{ relation_type }} {{ relation }} is {{ escaped_comment }}; #}
+  SELECT 1;
+{% endmacro %}
+
+{# CrateDB: `COMMENT ON` not supported. #}
+{% macro cratedb__alter_column_comment(relation, column_dict) %}
+  {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
+  {% for column_name in column_dict if (column_name in existing_columns) %}
+    {% set comment = column_dict[column_name]['description'] %}
+    {% set escaped_comment = postgres_escape_comment(comment) %}
+    {# comment on column {{ relation }}.{{ adapter.quote(column_name) if column_dict[column_name]['quote'] else column_name }} is {{ escaped_comment }}; #}
+    SELECT 1;
+  {% endfor %}
+{% endmacro %}
+
+{# TODO: Enable this, derived from `default__reset_csv_table`? #}
+{#
+{% macro cratedb__reset_csv_table(model, full_refresh, old_relation, agate_table) %}
+    {% set sql = "" %}
+    {% if full_refresh %}
+        {{ adapter.drop_relation(old_relation) }}
+        {% set sql = create_csv_table(model, agate_table) %}
+    {% else %}
+        {{ adapter.truncate_relation(old_relation) }}
+        {% set sql = "delete from " ~ old_relation %}
+    {% endif %}
+
+    {{ return(sql) }}
+{% endmacro %}
+#}
```

Comment:
I've derived this from @hlcianfagna's contributions at [Using dbt with CrateDB](https://community.cratedb.com/t/using-dbt-with-cratedb/1566). However, it has not been activated yet.

- https://github.com/crate/cratedb-examples/pull/754#discussion_r1862067486